docs/rfcs/XXXX_storage_backends.md

```diff
@@ -0,0 +1,288 @@
+
+#### Configuration
+
+Each study bucket needs to be configured with the correct security policies, encryption, replication, logging, and so on. For this reason, the bucket creation process was automated and included as part of the the application's service offerings. Upon creation, the the application will configure these additional settings ensuring all buckets are setup correctly.
+
+#### Layout
+Each study bucket has a similar layout:
+```
+source/
+  uploads/
+harmonized/
+```
+
+Under `source` are data files such as raw read files received from the sequencing center, MRI DICOMs from the investigator, and so forth. The `uploads` directory contains files that are uploaded from the the application. `harmonized` contains files that were processed by the BIX team using things such as a standard alignment workflow or calling pipeline.
+
+### Issues with Study Buckets
+
+Study buckets are restricted to the same account as the IAM profile assigned to the application's compute instance.
+The IAM profile contains a large surface of permissions needed to create and modify buckets.
+
+## Storage Backend Solution
+By allowing users to create "Storage Backends" that grant the application access to bucket resources in other accounts, we will be able to provide more flexibility in how organizations choose to store their data.
+
+## Default Storage
+![enter image description here](https://i.ibb.co/vBb1YsN/storage-layout.png)
+"Default Storage" will refer to the defaults for how a new organization and its studies will be stored. It's important that new organizations and studies be quickly bootstraped into a working state without unnecessary configuration. To achieve this, the application will provide a default workspace for documents to land. This will be achieved by providing a Storage Backend that will point to the application's application bucket.
+
+### Application Bucket
+
+It's proposed that the application maintain a single bucket in which it may stash data. This bucket will be intended to contain anything the the application may need to store such as configuration files, user avatars, and similar. Most notably, though, it will contain uploaded documents for each study. In this way, the single application bucket will deprecate the need for each study bucket to contain its uploads. However, it is also recognized that perhaps some organizations may wish their uploaded documents to reside within their own resource pool. For this case, we will allow the document bucket to be specified within a different account.
```

Comment:
```suggestion
It's proposed that the application maintain a single bucket in which it may stash data. This bucket will be intended to contain anything the application may need to store - such as configuration files, user avatars, etc. Most notably, though, it will contain uploaded documents for each study. In this way, the single application bucket will deprecate the need for each study bucket to contain its uploads. However, it is also recognized that perhaps some organizations may wish their uploaded documents to reside within their own resource pool. For this case, we will allow the document bucket to be specified within a different account.
```